TraineeTasks/RegexTasks.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace TraineeTasks
+{
+    public class RegexTasks
+    {
+        /// <summary>
+        /// Find words in a string that contain the letter h, followed by 0 or 1 letter u.
+        /// </summary>
+        /// <param name="str"></param>
+        /// <returns>Returns a collection of strings with found words.</returns>
+        public string[] FindWords(string str)
+        {
+            if (string.IsNullOrWhiteSpace(str))
+            {
+                throw new ArgumentException("The string cannot be null or empty string");
+            }
+
+            return Regex.Matches(str, @"\w*h[u]{0,1}[^u]\w*", RegexOptions.IgnoreCase)
```

Comment:
Are you sure that you want to use IgnoreCase option here?

Reply:
I removed it